zaza/model.py

```diff
@@ -725,32 +732,40 @@ def one_agent_status(model, status):
             lambda: len(model.units) > 0
         )
         logging.info("Waiting for all units to be idle")
-        await model.block_until(
-            lambda: model.all_units_idle(), timeout=timeout)
-        for application in model.applications:
-            check_info = states.get(application, {})
-            for unit in model.applications[application].units:
-                logging.info("Checking workload status of {}".format(
-                    unit.entity_id))
-                await model.block_until(
-                    lambda: check_unit_workload_status(
-                        model,
-                        unit,
-                        check_info.get('workload-status', 'active')),
-                    timeout=timeout)
-                check_msg = check_info.get('workload-status-message')
-                logging.info("Checking workload status message of {}".format(
-                    unit.entity_id))
-                if check_msg is not None:
-                    prefixes = (check_msg)
-                else:
-                    prefixes = approved_message_prefixes
-                await model.block_until(
-                    lambda: check_unit_workload_status_message(
-                        model,
-                        unit,
-                        prefixes=prefixes),
-                    timeout=timeout)
+        try:
+            await model.block_until(
+                lambda: model.all_units_idle(), timeout=timeout)
+        except concurrent.futures._base.TimeoutError as e:
```

Comment:
`e` hasn't been used here; it should fail pep8, and if not, then there is a problem with linting.